apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+  start_pool(lists:last(Pools), NodeToId),
+  {next_state, tasks, send_check(State)};
+
+tasks_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
+  State1 = State#state{pools = connect(Node, Pools), passive_pools = lists:delete(Node, PassivePools)},
+  {NextState, NewState} = choose_pool(State1),
+  {next_state, NextState, NewState};
+
+tasks_({pool_enabled, Node}, #state{pools = Pools} = State) ->
+  {NextState, NewState} = choose_pool(State#state{pools = enable(Node, Pools)}),
+  {next_state, NextState, NewState};
+
+tasks_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools, passive_pools = []} = State) ->
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  {next_state, tasks, State#state{retasks = NewReTasks, pools = delete(Node, Pools), passive_pools = [Node]}};
+
+tasks_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools,
+  passive_pools = PassivePools} = State) ->
+
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  NewState = State#state{retasks = NewReTasks, pools = delete(Node, Pools), passive_pools = [Node|PassivePools]},
+  {next_state, tasks, NewState};
+
+tasks_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  NewPools = disable(Node, Pools),
+  case status(Node, Pools) of
+    {pending, _Ref, Id} ->
+      case task(Category, Id) of
+        undefined ->
+          {next_state, tasks, State#state{pools = NewPools}};
+        Task ->
+          rpc:cast(Node, fyler_pool, cancel_task, [Id]),
+          {next_state, tasks, State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}}
+      end;
+    _ ->
+      {next_state, tasks, State#state{pools = NewPools}}
+  end;
+
+tasks_({task_accepted, Ref, Node}, #state{pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, _Id} ->
+      {NextState, NewState} = choose_pool(State#state{pools = enable(Node, Pools)}),
+      {next_state, NextState, NewState};
+    _ ->
+      {next_state, tasks, State}
+  end;
+
+tasks_({task_rejected, Ref, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, Id} ->
+      NewPools = enable(Node, Pools),
+      case task(Category, Id) of
+        undefined ->
+          {next_state, tasks, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools}),
+          {next_state, NextState, NewState}
+      end;
+    _ ->
+      {next_state, tasks, State#state{pools = Pools}}
+  end;
+
+tasks_({run, #task{priority = Priority} = Task}, #state{tasks = Tasks, priorities = Priorities} = State) ->
+  update_ets(Task),
+  NumPriority = maps:get(Priority, Priorities, 1),
+  {next_state, tasks, State#state{tasks = fyler_queue:in(Task, NumPriority, Tasks)}};
```

Comment:
Line 423 is too long:   {next_state, tasks, State#state{tasks = fyler_queue:in(Task, NumPriority, Tasks)}};.